include/cgimap/backend/apidb/changeset_upload/node_updater.hpp

```diff
@@ -20,11 +20,14 @@
 #include <memory>
 #include <set>
 
+struct RequestContext;
+
 
 class ApiDB_Node_Updater : public api06::Node_Updater {
 
 public:
   ApiDB_Node_Updater(Transaction_Manager &_m,
+                     const RequestContext& req_ctx,
```

Comment:
I've noticed that you're passing the new request context to {node,way,relation}_updater, where we previously didn't pass any of the information. Did you have some specific use cases in mind where this might come in handy? Not really a big issue, just wanted to understand the motivation behind this change.

Reply:
I don't think a simple bounding box size check will be enough to combat vandalism and we could certainly use more tools in our toolbox for that purpose. Providing the request context here where the uploaded objects are processed allows for checks and filters that take the actual map data into account (like e.g. a word filter for tags, region or object specific protections, etc...)
As I mentioned over in https://github.com/zerebubuth/openstreetmap-cgimap/pull/409#issuecomment-2171913919 I'm currently tinkering with a framework for checks like that which would require this information. [^1] It will certainly still take a while though until (/if) that is complete and I can carry these changes in my own branch anyway, so having these changes upstreamed is not really necessary at the moment if they aren't needed immediately.

[^1]: Although the biggest hurdle there is that we never actually have a complete look at the uploaded data since all changes are pushed to the database in chunks as soon as they are parsed.

Reply:
I also have some doubts that the bounding box checks will be sufficient in the end, or if it would even cause too much collateral damage. The approach I started looking into is to throw a list of created/modified/deleted object over the fence, and let some db function do the impact analysis, like what is total way length, how much displacement do we see. 

We could leverage PostGIS to do all sorts of geo analysis, and even include ML stuff through the plpython3u extension. These are all theoretical options, and it highly depends on what operations thinks is reasonable. 

Since we both see the to-be committed data, as well as the previous data on the database, chunked processing would not so much of an issue. I also want to keep the CGImap code clean and would try to keep complicated analysis logic outside the project as much as possible.

That's just my opinion. It's good to have different alternative ideas out there, and see what works best. Somehow we also need to consolidate all the requirements and then decide what data points would be needed in each case. This might help us to avoid too many "black swan" like situations in the future.